browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -300,11 +300,19 @@ void SpeedreaderTabHelper::ProcessNavigation(
   TransitStateTo(DistillStates::DistillReverting(reason, false), true);
   TransitStateTo(DistillStates::ViewOriginal(), true);
 
-  if (url_looks_readable && enabled_for_site) {
-    // Speedreader enabled for this page.
-    TransitStateTo(DistillStates::Distilling(
-                       DistillStates::Distilling::Reason::kAutomatic),
-                   true);
+  if (enabled_for_site) {
+    // Enable speedreader if the user explicitly enabled speedreader on the
+    // site.
+    const bool explicit_enabled_for_size =
+        kSpeedreaderExplicitPref.Get() &&
```

Comment:
why is this here instead of in `GetEnabledForSiteSetting`? Also `GetEnabledForSiteSetting` returns a ContentSetting so this doesn't seem right

Reply:
also shouldn't this use `IsExplicitlyEnabledForSite`?

Reply:
Yep, it is a mistake here, it should be `IsExplicitlyEnabledForSite`. it worked because `CONTENT_SETTING_DEFAULT` == 0 and it means that there is no explicit setting.